tests/tests.js

```diff
@@ -720,86 +744,210 @@ QUnit.test("validateForm test", function(assert) {
+    this.helloState = {uniqueId: '1234', title: 'hello', startDateString: "20170101", endDateString: "20171201"};
+    this.worldState = {uniqueId: '5678', title: 'world', startDateString: "20160101", endDateString: "20161201"};
+    localStorage.setItem(uniqueId, JSON.stringify(this.state));
+    localStorage.setItem(this.helloState.uniqueId, JSON.stringify(this.helloState));
+    localStorage.setItem(this.worldState.uniqueId, JSON.stringify(this.worldState));
+    //add loading wheel
+    this.$calendarDiv.append(`<div id="loadingWheel"></div>`);
+    
   },
   afterEach: function() {
     // clean up after each test
-    this.$calendarDiv.empty();
+    this.$calendarDiv.empty()
   }
 });
 
-// TODO test that tests if the correct init function runs depending on
-// which storage is passed to CheckIt
-QUnit.test("displayActiveCalendar test", function(assert) {
-    assert.expect(0);
-});
 
-QUnit.test("fillDropdown test", function(assert) {
-    assert.expect(0);
-});
-
-QUnit.test("initLocalStorage test", function(assert) {
-    assert.expect(0);
-    console.log(typeof(this.checkit));
+QUnit.test("displayActiveCalendar test", function(assert) {
+    assert.expect(8);
+    var $calendarDiv = this.$calendarDiv;
+    
+    // When there is a current Active Calendar
+    assert.equal($calendarDiv.find('h1').text(), 'Test Calendar');
+    // currentActiveCalendar
+    var done = assert.async(2);
+    assert.equal($calendarDiv.find('.monthframe').length, 1);
+    var displayActiveP = this.checkit.displayActiveCalendar();
+    displayActiveP.then(function () {
+        assert.equal($calendarDiv.find('h1').text(), 'hello');
+        assert.equal($calendarDiv.find('.monthframe').length, 12);
+        done()
+    });
     
+    // When there isn't a current active calendar
+    $calendarDiv.empty();
+    assert.equal($calendarDiv.find('.monthframe').length, 0);
+    localStorage.removeItem('current_active_calendar');
+    assert.equal(JSON.parse(localStorage.getItem('current_active_calendar')), null);
+    var displayNoP = this.checkit.displayActiveCalendar();
+    displayNoP.then(function(val) {
+        assert.equal(val, "Not found");
+        assert.equal($calendarDiv.find('.monthframe').length, 0);
+        done()
+    });
+
 });
 
 QUnit.test("onActivityChanged test", function(assert) {
-    assert.expect(0);
+    assert.expect(3);
+    // check that loading wheel is displayed when active calls > 0 and hidden
+    // when active calls === 0, what about if active calls < 0? throw an error?
+    
+    //active calls > 0
+    assert.equal($('#loadingWheel').children().length, 0);
+    this.checkit.onActivityChanged(1, 'loadingWheel');
+    assert.equal($('#loadingWheel').children().length, 1);
+    
+    // active calls === 0
+    this.checkit.onActivityChanged(0, 'loadingWheel');
+    assert.equal($('#loadingWheel').children().length, 0);
+    
+    // it's possible for active calls to be less than zero, maybe I should have the loading wheel
```

Comment:
This is the kind of error that you don't really want to "fix" because it means your UI is going to be doing wrong/bad things. For example, if the count is negative and I start doing work, then the spinner won't appear because the count is not positive.

I'd assert that active calls is >= 0 whenever you decrement it.